cs4660/graph/graph.py

```diff
@@ -86,22 +122,53 @@ def __init__(self):
         self.adjacency_list = {}
 
     def adjacent(self, node_1, node_2):
-        pass
+        if node_1 not in self.adjacency_list or node_2 not in self.adjacency_list:
+            return False
+        for e in self.adjacency_list[node_1]:
+            if node_2 == e.to_node:
+                return True
+        for e in self.adjacency_list[node_2]:
+            if node_1 == e.to_node:
+                return True
```

Comment:
second check here is unnecessary and it will make the adjacent method to be checking adjacent for undirect graph rather than direct graph.

In other word, you are checking the direction from node_1 to node_2 or node_2 to node_1, which is wrong based on the adjacent definition.